src/index.ts

```diff
@@ -349,7 +359,7 @@ function createServicePrincipal(azureWebsiteName: string, subscriptionId: string
                 endDate,
                 keyId: uuid.v1(),
                 startDate,
-                value: servicePrincipalSecret
+                value: newServicePrincipalSecret
```

Comment:
Should this also have a check for existingServicePrincipalSecret ?

existingServicePrincipalSecret ? existingServicePrincipalSecret : newServicePrincipalSecret

Reply:
No this is used for the new application where as the existing service principal value is passed into the arm template if provided by the user. In case they have admin/owner access on the sub and dont provide service principal secret we can use newServicePrincipalSecret to pass to the arm template.